### PR TITLE
Perform line-search inside a while loop to ensure that the criterion is always met.

### PR DESCRIPTION
**Pull request by fabianp**

### CODE REVIEW

File: lightning/impl/sag_fast.pyx [hunk 1]
```diff
@@ -322,14 +322,16 @@ def _sag_fit(self,
...
                         square_norm_X[i] += (data[j] * data[j])
-                a = loss.loss(y_pred - g[i] * square_norm_X[i] / lipschitz, y[i])
-                b = loss.loss(y_pred, y[i]) - 0.5 * square_norm_X[i] * (g[i] * g[i]) / lipschitz
-                if a <= b :
-                    # condition is satisfied, decrease Lipschitz constant
-                    lipschitz /= line_search_scaling
-                else:
-                    # condition not satisfied, decrease step size
-                    lipschitz *= 2.0
+                while True:
```
Comment (mblondel) [inline lightning/impl/sag_fast.pyx]: I would set a max number of iterations for safety.

So, why did you decide to implement a loop in the end?

Comment (fabianp) [inline lightning/impl/sag_fast.pyx]: I got NaNs on an exponential loss function. This happened on the first iteration, a possibility which i had not considered.

**Comment by fabianp** [comment]
Added maximum number of iterations for line-search.

**Comment by fabianp** [comment]
good to merge now @mblondel ?

**Comment by mblondel** [comment]
Yes, please merge :) I just assumed you were still working on this.

**Comment by fabianp** [comment]
great, thanks

**Pull request merged**